src/main/java/uk/co/onsdigital/job/persistence/DataSetRepository.java

```diff
@@ -6,14 +6,24 @@
 import uk.co.onsdigital.job.exception.NoSuchDataSetException;
 
 import javax.persistence.*;
-import java.util.UUID;
+import java.util.*;
+import java.util.stream.Collectors;
 
 /**
  * Repository for looking up datasets.
  */
 @Component
 public class DataSetRepository {
 
+    /** Query that returns all existing dimensions that match the requested dimensions for a dataset. */
+    public static final String DIMENSION_VALUES_QUERY = "SELECT dim.dimension.name, dim.value FROM DimensionValue dim " +
+            "WHERE dim.dimension.dataSet.id = :ddsId " +
+            "AND dim.dimension.name IN :names " +
+            "AND dim.value IN :values";
```

Comment:
I think this can match values that exist in one of the requested dimensions, but not necessarily the same dimension that the requested value is for? For example if we request `{"foo": ["a"], "bar": ["b"]}` but in the database there is a `foo` dimension with a `b` value and a `bar` dimension with an `a` value then this would match even though the values are in the wrong dimensions?